resources/assets/less/bem/grid.less

```diff
@@ -0,0 +1,37 @@
+/**
+ *    Copyright 2015-2017 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+.grid {
+  // 840px seems a big large for this to trigger?
+  @media @desktop {
+    display: flex;
+    flex-wrap: wrap;
+  }
+
+  // Use this if we don't want the grid to collapse into a stack.
+  // FIXME: better name
+  &--xs {
+    display: flex;
+  }
+
+  &--gutters {
+    margin-left: -@flex-grid--gutter-width;
```

Comment:
Top and bottom spacing in grid column layouts tend to be different from the column gutters, so we don't really want to apply them together.